### PR TITLE
Use cp -u on linux-musl

### PR DESCRIPTION
**Pull request by kasper3**
This actually applies to all busybox based OS. Fixes https://github.com/dotnet/cli/issues/8738.
Also updated readme: alpine-<ver>-targz to musl-targz. Discussion: https://github.com/dotnet/cli/commit/65d5730631ef3cae8435be66b37b6c03b49d5dbf#r28750312

@weshaggard, @johnbeisner 

Follow-up to: https://github.com/dotnet/cli/pull/9104

### CODE REVIEW

**Comment by kasper3** [comment]
test RHEL7.2 x64 Release Build

File: README.md [hunk 1]
```diff
@@ -123,8 +123,8 @@ Reference notes:
 [rhel-6-targz]: https://dotnetcli.blob.core.windows.net/dotnet/Sdk/release/2.1.3xx/dotnet-sdk-latest-rhel.6-x64.tar.gz
 [rhel-6-targz-checksum]: https://dotnetclichecksums.blob.core.windows.net/dotnet/Sdk/release/2.1.3xx/dotnet-sdk-latest-rhel.6-x64.tar.gz.sha
 
-[alpine-3.6-targz]: https://dotnetcli.blob.core.windows.net/dotnet/Sdk/release/2.1.3xx/dotnet-sdk-latest-alpine.3.6-x64.tar.gz
-[alpine-3.6-targz-checksum]: https://dotnetclichecksums.blob.core.windows.net/dotnet/Sdk/release/2.1.3xx/dotnet-sdk-latest-alpine.3.6-x64.tar.gz.sha
+[linux-musl-targz]: https://dotnetcli.blob.core.windows.net/dotnet/Sdk/release/2.1.3xx/dotnet-sdk-latest-alpine.3.6-x64.tar.gz
+[linux-musl-targz-checksum]: https://dotnetclichecksums.blob.core.windows.net/dotnet/Sdk/release/2.1.3xx/dotnet-sdk-latest-alpine.3.6-x64.tar.gz.sha
```
Comment (johnbeisner) [inline README.md]: @kasper3 
Change "alpine.3.6" to "linux-musl" in the actual link.

Comment (kasper3) [reply README.md]: Fixed

**Comment by livarcocc** [comment]
Marking was WIP until the branches are open for RTM.

**Comment by livarcocc** [comment]
Thanks for the PR @kasper3

**Comment by kasper3** [comment]
Thanks, I am able to get 2.1 on fresh install of Alpine:

```sh
apk update

apk add bash curl git libunwind-dev libintl openssl-dev icu-libs curl-dev

curl -O https://raw.githubusercontent.com/dotnet/cli/release/2.1.3xx/scripts/obtain/dotnet-install.sh

chmod 777 dotnet-install.sh

./dotnet-install.sh -Channel release/2.1.3xx

export PATH=$PATH:/root/.dotnet
```

then `dotnet new ....` etc.

**Pull request merged**